examples/sampling/graphbolt/link_prediction.py

```diff
@@ -78,6 +78,40 @@ def forward(self, blocks, x):
                 hidden_x = F.relu(hidden_x)
         return hidden_x
 
+    def inference(self, graph, features, dataloader, device):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        feature = features.read("node", None, "feat")
+
+        buffer_device = torch.device("cpu")
```

Comment:
Get this from feature fetcher, as we will have GPU feature fecther in the near future.

Reply:
If we get the device from feature fetcher (equal to `args.device`) the `pin_memory` will be useless?